### PR TITLE
Turael Slayer List

### PR DESCRIPTION
**Pull request by SacredClay**

### CODE REVIEW

**Comment by Alexsuperfly** [comment]
can you at least attempt to compile this before PRing it

**Comment by SacredClay** [comment]
> can you at least attempt to compile this before PRing it

File: src/monsters/turael/CaveSlime.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+import { GemTable } from '../subtables/RareDropTable';
+
+export const CaveSlimeTable = new LootTable()
+	.every('Swamp tar', [1, 6])
+	.tertiary(128, 'Clue scroll (easy)', 1)
```
Comment (wyattos) [inline src/monsters/turael/CaveSlime.ts]: .tertiary(128, 'Clue scroll (easy)')

File: src/monsters/turael/CaveSlime.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	.addItem('Bronze axe', 1, 16)
+	.addItem('Iron sword', 1, 12)
+	.addItem('Bronze full helm', 1, 14)
+	.addItem('Iron kiteshield', 1, 10)
+	.addItem('Iron boots', 1, 1)
+	/*Runes */
+	.addItem('Water rune', 15, 20)
+	.addItem('Earth rune', 5, 7)
+	/*Other */
+	.addItem('Coins', 1, 30)
```
Comment (wyattos) [inline src/monsters/turael/CaveSlime.ts]: remove all the coin drops and replace it with a table
	.add( 
		new LootTable() 
			.add('Coins', 1),
			.add('Coins', 4),
			.add('Coins', 10), 
			.add('Coins', 22), 
			.add('Coins', 46)
		x 
	)

File: src/monsters/turael/Cow.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+
+export const CowTable = new LootTable()
+	.every('Bones')
+	.every('Cowhide')
+	.every('Raw beef')
+	.tertiary(128, 'Clue scroll (beginner)', 1);
```
Comment (wyattos) [inline src/monsters/turael/Cow.ts]: .tertiary(128, 'Clue scroll (beginner)');

File: src/monsters/turael/CrawlingHand.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	.addItem('Yellow gloves', 1, 2)
+	.addItem('Red gloves', 1, 2)
+	.addItem('Teal gloves', 1, 2)
+
+	/* Rings */
+	.addItem('Gold ring', 1, 3)
+	.addItem('Sapphire ring', 1, 2)
+	.addItem('Emerald ring', 1, 2)
+
+	/* Coins */
```
Comment (wyattos) [inline src/monsters/turael/CrawlingHand.ts]: see cave slime comment for how to fix this

File: src/monsters/turael/CrawlingHand.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	.addItem('Coins', 5, 10)
+	.addItem('Coins', 8, 8)
+	.addItem('Coins', 10, 8)
+
+	/* Subtables */
+
+	.add(GemTable, 4)
+
+	/* Tertiary */
+	.tertiary(500, 'Crawling hand', 1);
```
Comment (wyattos) [inline src/monsters/turael/CrawlingHand.ts]: .tertiary(500, 'Crawling hand');

File: src/monsters/turael/Dog.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+
+export const DogTable = new LootTable().every('Bat bones').tertiary(25, 'Ensouled dog head', 1);
```
Comment (wyattos) [inline src/monsters/turael/Dog.ts]: fix bat bones,
.tertiary(25, 'Ensouled dog head');

Comment (wyattos) [inline src/monsters/turael/Dog.ts]: GuardDogTable

File: src/monsters/turael/Dwarf.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+import { GemTable } from '../subtables/RareDropTable';
+
+export const DwarfTable = new LootTable(128)
+	.every('Bones')
+	.tertiary(100, 'Clue scroll (beginner)', 1)
+	/* Weapons and Armour*/
```
Comment (wyattos) [inline src/monsters/turael/Dwarf.ts]: leave a space above weapons and armor

File: src/monsters/turael/Dwarf.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+	.addItem('Coins', 30, 2)
+
+	/* Other */
+	.addItem('Hammer', 1, 10)
+	.addItem('Bronze bar', 1, 7)
+	.addItem('Iron ore', 1, 4)
+	.addItem('Tin ore', 1, 3)
+	.addItem('Copper ore', 1, 3)
+	.addItem('Iron bar', 1, 3)
+	.addItem('Coal', 1, 3)
```
Comment (wyattos) [inline src/monsters/turael/Dwarf.ts]: weight of 2

File: src/monsters/turael/Dwarf.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+import { GemTable } from '../subtables/RareDropTable';
+
+export const DwarfTable = new LootTable(128)
+	.every('Bones')
+	.tertiary(100, 'Clue scroll (beginner)', 1)
```
Comment (wyattos) [inline src/monsters/turael/Dwarf.ts]: .tertiary(100, 'Clue scroll (beginner)')

File: src/monsters/turael/Ghost.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+
+export const GhostTable = new LootTable().tertiary(90, 'Clue scroll (beginner)', 1);
```
Comment (wyattos) [inline src/monsters/turael/Ghost.ts]: .tertiary(90, 'Clue scroll (beginner)');

File: src/monsters/turael/Icefiend.ts [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+import { GemTable } from '../subtables/RareDropTable';
+
+export const IcefiendTable = new LootTable(128)
+	.tertiary(128, 'Clue scroll (easy)', 1)
```
Comment (wyattos) [inline src/monsters/turael/Icefiend.ts]: .tertiary(128, 'Clue scroll (easy)')

File: src/monsters/turael/Icefiend.ts [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+import { GemTable } from '../subtables/RareDropTable';
+
+export const IcefiendTable = new LootTable(128)
```
Comment (wyattos) [inline src/monsters/turael/Icefiend.ts]: currently the total weights on the table is 37, this means youll get 1 drop every 3.4 kills. i'd change this lower to like ~50 or maybe even 37 to guarantee a drop every kill

File: src/monsters/turael/Lizard.ts [hunk 13]
```diff
@@ -0,0 +1,67 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+import { GemTable } from '../subtables/RareDropTable';
+
+export const LizardTable = new LootTable()
+	.every('Big bones')
```
Comment (wyattos) [inline src/monsters/turael/Lizard.ts]: leave a space above armour

File: src/monsters/turael/Lizard.ts [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+	/* Armour */
+	.oneIn(512, 'Mystic gloves (light)', 1)
```
Comment (wyattos) [inline src/monsters/turael/Lizard.ts]: .oneIn(512, 'Mystic gloves (light)')

File: src/monsters/turael/Lizard.ts [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+
+	/* Runes */
+	.addItem('Fire rune', [5, 42], 25)
+	.addItem('Nature rune', 5, 1)
+
+	/* Herbs */
```
Comment (wyattos) [inline src/monsters/turael/Lizard.ts]: .add(HerbDropTable, 1, x)

File: src/monsters/turael/Lizard.ts [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+	/* Other */
+
+	.addItem('Kebab', 1, 19)
+	.addItem('Knife', 1, 24)
+	.addItem('Waterskin(0)', 2, 6)
+	.addItem('Waterskin(1)', 2, 6)
+	.addItem('Waterskin(2)', 2, 6)
+	.addItem('Waterskin(3)', 2, 6)
+	.addItem('Waterskin(4)', 2, 6)
+	/* Tertiary */
```
Comment (wyattos) [inline src/monsters/turael/Lizard.ts]: remove the 1 at the end of both tertiaries

File: src/monsters/turael/Lizard.ts [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+	.tertiary(400, 'Long bone', 1)
+	.tertiary(5012.5, 'Curved bone', 1)
```
Comment (wyattos) [inline src/monsters/turael/Lizard.ts]: this has to be an integer ie 5013

File: src/monsters/turael/Lizard.ts [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+
+	/* Subtables */
+	.add(GemTable, 1, 42);
```
Comment (wyattos) [inline src/monsters/turael/Lizard.ts]: weight is extremely high it should be much lower

File: src/monsters/turael/Minotaur.ts [hunk 19]
```diff
@@ -0,0 +1,39 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+import { GemTable } from '../subtables/RareDropTable';
+
+export const MinotaurTable = new LootTable(101)
+	.every('Bones')
+	.tertiary(50, 'Ensouled minotaur head', 1)
```
Comment (wyattos) [inline src/monsters/turael/Minotaur.ts]: remove the 1's at the end of tertiary

File: src/monsters/turael/Minotaur.ts [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+	/* Runes */
+	.addItem('Mind rune', 1, 6)
+
+	/* Other */
+	.addItem('Coins', [2, 84], 16)
+	.addItem('Tin ore', 1, 13)
+	.addItem('Rune essence', 5, 6)
+	.addItem('Cooked meat', 1, 7)
+	.addItem('Copper ore', 1, 5)
+	.addItem('Pure essence', 15, 6)
```
Comment (wyattos) [inline src/monsters/turael/Minotaur.ts]: see above on how to fix pure essence

File: src/monsters/turael/Minotaur.ts [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+	.addItem('Pure essence', 20, 4)
+	.addItem('Rune essence', 5, 6)
+
+	/* Subtables */
+	.add(GemTable, 1, 20);
```
Comment (wyattos) [inline src/monsters/turael/Minotaur.ts]: weight is too high

File: src/monsters/turael/Scorpion.ts [hunk 22]
```diff
@@ -0,0 +1,12 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+
+export const ScorpionTable = new LootTable(25)
+	.addItem('Ensouled scorpion head', 1, 1)
```
Comment (wyattos) [inline src/monsters/turael/Scorpion.ts]: tertiary for the head

File: src/monsters/turael/Scorpion.ts [hunk 23]
```diff
@@ -0,0 +1,12 @@
...
+	.tertiary(100, 'Clue scroll (beginner)', 1);
```
Comment (wyattos) [inline src/monsters/turael/Scorpion.ts]: remove 1 at the end of tertiary

File: src/monsters/turael/Skeleton.ts [hunk 24]
```diff
@@ -0,0 +1,54 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+
+export const SkeletonTable = new LootTable(128)
+	.every('Bones')
+	.tertiary(5000, 'Skeleton champion scroll', 1)
```
Comment (wyattos) [inline src/monsters/turael/Skeleton.ts]: remove 1

File: src/monsters/turael/Skeleton.ts [hunk 25]
```diff
@@ -0,0 +1,54 @@
...
+
+	/* Herbs */
+
+	.addItem('Grimy guam leaf', 1, 6)
```
Comment (wyattos) [inline src/monsters/turael/Skeleton.ts]: .add(HerbDropTable, 1, x)

File: src/monsters/turael/Skeleton.ts [hunk 26]
```diff
@@ -0,0 +1,54 @@
...
+	.addItem('Grimy ranarr weed', 1, 5)
+	.addItem('Grimy irit leaf', 1, 5)
+	.addItem('Grimy avantoe', 1, 3)
+	.addItem('Grimy kwuarm', 1, 3)
+	.addItem('Grimy cadantine', 1, 3)
+	.addItem('Grimy lantadyme', 1, 3)
+	.addItem('Grimy dwarf weed', 1, 3)
+
+	/* Runes/Ammunition */
+
```
Comment (wyattos) [inline src/monsters/turael/Skeleton.ts]: delete this space

File: src/monsters/turael/Skeleton.ts [hunk 27]
```diff
@@ -0,0 +1,54 @@
...
+	.addItem('Bronze arrow', 2, 20)
```
Comment (wyattos) [inline src/monsters/turael/Skeleton.ts]: see above on how to fix bronze arrows

File: src/monsters/turael/Skeleton.ts [hunk 28]
```diff
@@ -0,0 +1,54 @@
...
+	.addItem('Fire rune', 2, 4)
+	.addItem('Nature rune', 3, 4)
+	.addItem('Iron arrow', 1, 4)
+	.addItem('Steel arrow', 1, 3)
+
+	/* Weapons */
+	.addItem('Iron dagger', 1, 8)
+
+	/* Other */
+	.addItem('Coins', 2, 24)
```
Comment (wyattos) [inline src/monsters/turael/Skeleton.ts]: see above on how to fix coins

File: src/monsters/turael/Skeleton.ts [hunk 29]
```diff
@@ -0,0 +1,54 @@
...
+	.addItem('Coins', 4, 24)
+	.addItem('Coins', 12, 24)
+	.addItem('Coins', 16, 24)
+	.addItem('Coins', 25, 20)
+	.addItem('Coins', 33, 20)
+	.addItem('Fire talisman', 1, 10)
+	.tertiary(100, 'Clue scroll (beginner)', 1)
```
Comment (wyattos) [inline src/monsters/turael/Skeleton.ts]: remove 1's

File: src/monsters/turael/Spider.ts [hunk 30]
```diff
@@ -0,0 +1,10 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+
+export const SpiderTable = new LootTable().tertiary(128, 'Clue scroll beginner', 1);
```
Comment (wyattos) [inline src/monsters/turael/Spider.ts]: remove 1

File: src/monsters/turael/Wolf.ts [hunk 31]
```diff
@@ -0,0 +1,12 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+
+export const WolfTable = new LootTable()
+	.every('Wolf bones')
+	.tertiary(128, 'Clue scroll (beginner)', 1);
```
Comment (wyattos) [inline src/monsters/turael/Wolf.ts]: remove 1

File: src/monsters/turael/Zombie.ts [hunk 32]
```diff
@@ -0,0 +1,41 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+import HerbDropTable from '../subtables/HerbDropTable';
+
+export const ZombieTable = new LootTable(128)
+	.every('Bones')
+	.tertiary(5000, 'Zombie champion scroll', 1)
```
Comment (wyattos) [inline src/monsters/turael/Zombie.ts]: remove 1

File: src/monsters/turael/Zombie.ts [hunk 33]
```diff
@@ -0,0 +1,41 @@
...
+	/* Weapons and armour */
```
Comment (wyattos) [inline src/monsters/turael/Zombie.ts]: add a space above weapons and armor

File: src/monsters/turael/Zombie.ts [hunk 34]
```diff
@@ -0,0 +1,41 @@
...
+	.addItem('Bronze med helm', 1, 4)
+	.addItem('Bronze longsword', 1, 1)
+	.addItem('Iron axe', 1, 1)
+	/*Runes and ammunition*/
```
Comment (wyattos) [inline src/monsters/turael/Zombie.ts]: add a space above

File: src/monsters/turael/Zombie.ts [hunk 35]
```diff
@@ -0,0 +1,41 @@
...
+	.addItem('Iron arrow', 5, 7)
+	.addItem('Body rune', 6, 5)
+	.addItem('Mind rune', 5, 5)
+	.addItem('Air rune', 13, 4)
+	.addItem('Iron arrow', 8, 4)
+	.addItem('Steel arrow', 5, 2)
+	.addItem('Nature rune', 6, 1)
+
+	/*Coins*/
+
```
Comment (wyattos) [inline src/monsters/turael/Zombie.ts]: remove space

**Pull request closed without merging**